airbyte-integrations/bases/standard-destination-test/src/main/java/io/airbyte/integrations/standardtest/destination/comparator/AdvancedTestDataComparator.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.standardtest.destination.comparator;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AdvancedTestDataComparator extends BasicTestDataComparator {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AdvancedTestDataComparator.class);
+
+  public static final String AIRBYTE_DATE_FORMAT = "yyyy-MM-dd";
+  public static final String AIRBYTE_DATETIME_FORMAT = "yyyy-MM-dd'T'HH:mm:ss";
+  public static final String AIRBYTE_DATETIME_WITH_TZ_FORMAT = "yyyy-MM-dd'T'HH:mm:ssXXX";
+
+  @Override
+  public void assertSameData(List<JsonNode> expected, List<JsonNode> actual) {
+    LOGGER.info("Expected data {}", expected);
+    LOGGER.info("Actual data   {}", actual);
+    assertEquals(expected.size(), actual.size());
+    final Iterator<JsonNode> expectedIterator = expected.iterator();
+    final Iterator<JsonNode> actualIterator = actual.iterator();
+    while (expectedIterator.hasNext() && actualIterator.hasNext()) {
+      compareObjects(expectedIterator.next(), actualIterator.next());
+    }
+  }
+
+  protected void compareObjects(final JsonNode expectedObject, final JsonNode actualObject) {
+    LOGGER.info("Expected Object : {}", expectedObject);
+    LOGGER.info("Actual Object   : {}", actualObject);
+    if (areBothEmpty(expectedObject, actualObject)) {
+      LOGGER.info("Both rows are empty.");
+    } else {
+      final Iterator<Map.Entry<String, JsonNode>> expectedDataIterator = expectedObject.fields();
+      assertEquals(expectedObject.size(), actualObject.size(), "Unequal row size");
+      while (expectedDataIterator.hasNext()) {
+        final Map.Entry<String, JsonNode> expectedEntry = expectedDataIterator.next();
+        final JsonNode expectedValue = expectedEntry.getValue();
+        JsonNode actualValue = null;
+        String key = expectedEntry.getKey();
+        for (final String tmpKey : resolveIdentifier(expectedEntry.getKey())) {
+          actualValue = actualObject.get(tmpKey);
+          if (actualValue != null) {
+            key = tmpKey;
+            break;
+          }
+        }
+        LOGGER.info("For {} Expected {} vs Actual {}", key, expectedValue, actualValue);
+        assertTrue(actualObject.has(key));
+        assertSameValue(expectedValue, actualValue);
+      }
+    }
+  }
+
+  private boolean areBothEmpty(final JsonNode expectedData, final JsonNode actualData) {
+    if (expectedData.size() == 0 && actualData.size() == 0) {
```

Comment:
nitpick: if we want to keep the `areBothEmpty` style, I think this branch is unnecessary? Since the next branch returns `(expectedData.size() == 0  || ...) && (actualData.size() == 0 || ...)`